[28기 정해수] 추가 구현 사항 진행중

wjdgotn77 opened the pull request:
## :: 최근 작업 주제 (하나 이상의 주제를 선택해주세요.)
- [x] 기능 추가
- [ ] 리뷰 반영
- [x] 리팩토링
- [ ] 버그 수정
- [ ] 컨벤션 수정

<br />

## :: 구현 목표 (해당 브랜치에서 구현하고자 하는 하나의 목표를 설정합니다.)
- 로그인 페이지 유효성검사 기능 구현.
- 메인 페이지 댓글 추가 및 삭제 기능 구현.

<br />

## :: 구현 사항 설명 (작업한 내용을 상세하게 기록합니다.)
### 1. 로그인 페이지
 -  input 태그를 사용해 사용자의 메일과 비밀번호를 입력할 수 있는 폼 작성.
 -  id 와 password 가 입력되면 로그인 버튼의 disabled가 해제.
 -  id 와 password 를 모두 입력한 뒤 enter 및 button 클릭으로 main.html로 이동.

[login.js] - id 및 password 유효성 검사 로직 추가
 id : 입력값 내에 "@"가 포함되지 않았다면 텍스트의 색상이 빨간색이고 유효한 이메일 입력을 요구하는 메세지 출력함수 추가.
 password : 입력값의 길이가 12자리보다 적다면 12자리 이상을 요구하는 메세지 출력 함수 추가.
 loginButton: 위의 두가지 조건이 충족되었을 때, 로그인 버튼이 활성화되며 엔터 및 버튼 클릭으로 main.html로 이동

### 2. 메인 페이지
 -  각 섹션 별로  html을 작성해주고 페이지 레이아웃 구성.
 -  피드 내 댓글 입력 창에 댓글을 입력하고 enter 또는 버튼 클릭시 댓글 추가 기능 구현.
 -  피드 내 댓글 입력 창에 댓글이 없다면 이벤트 미발생으로 댓글 추가 불가능.

[main.js] - 댓글 추가 및 삭제 기능 구현
 댓글창에 텍스트 입력후 엔터 및 버튼 클릭으로 댓글 추가할 수 있는 함수 추가.
 댓글 추가 시 오른쪽에 댓글을 삭제할 수 있는 x버튼 추가.
 x 버튼 클릭 시 현재의 댓글을 삭제할 수 있도록 delete 함수 구현.
 
<br />

## :: 성장 포인트 (해당 기능을 구현하며 고민했던 사항이나 새로 알게된 부분, 어려웠던 점 등을 작성합니다.)
### [어려웠던 부분]
1. 변수 및 함수의 네이밍 작성과 html의 클래스 네이밍 작성이 어려웠습니다.

2. 메인 페이지 css 작성시 `flex` 와 `position`을 이용해 레이아웃을 구성하는 부분이 어려웠습니다. 오른쪽 사이드에 유저의 프로필 및 추천팔로우 섹션을 해당 위치에 고정시키는 부분에서 초기 레이아웃을 잘못잡아 수정하는데에 오래 걸렸습니다.

3. 메인 페이지의 유저 아이디 검색 기능 구현 시 해당 검색 문자가 뜰 때마다 일치하는 아이디가 한번이 아닌 여러번 list에 출력되었는데, 해당 요소를 remove()로 지워보기도 하였는데 원하는 대로 동작하지 않아, 이부분에 대한 처리가 어려웠습니다.

### [새로 알게된 부분]
상단 네비게이션의 검색바에서 `position: reletive` 로 `input box` 내에 icon 을 넣어주는 대신, `div`로 래퍼를 만들어 주고 `flex`를 이용해주고, icon과 `input`을 넣어준 뒤 `input`의 속성 값을 unset 해주어 작성하였습니다. 이후 요소들을 추가해주거나, 대체해주어야 할 때 확장성을 고려해보았습니다.

<br />

## :: 기타 질문 및 특이 사항
추가 기능 구현 과제 중 아이디 검색기능을 구현시에 현재는 주석의 내용처럼
이벤트가 발생할 때마다 innerHTML을 비워주고 다시 그려주는 방식을 사용했는데,
텍스트 입력이 발생할 때마다 해당 로직이 발생하는 것이 비효율적일 수도 있겠다는 생각이 듭니다. 
다른 방식으로 리팩토링해본다면 어떤 방향으로 생각해볼 수 있을지 궁금합니다.

```js
// input 태그의 input 이벤트 함수
function handleSearch() {
  const searchContainer = document.querySelector(".search-container");
  const searchingId = searchInput.value;

  if (searchInput.value) {
    searchContainer.classList.remove("hide");
    checkedSearchId(searchingId);
  } else {
    searchContainer.classList.add("hide");
   // 현재의 위치에서 ul 태그를 remove() 했을때, 한번 지워진 이후 다시 li가 그려지지 않아서 원하는 대로 동작하지 않았습니다.
  }
}

// 아이디를 검색해주는 함수
function checkedSearchId(word) {
  // 이벤트가 발생할 때마다 innerHTML을 비워주고 다시 그려준다.
  const searchList = document.querySelector(".search-list");
  searchList.innerHTML = "";
  idList.forEach((info) => {
    if (info.id.includes(word)) {
      const li = document.createElement("li");
      const liContents = `
        <img alt="profile image"
          src=${info.src}
          class="search-image"
        />
        <div class="search-profile">
          <p class="search-id">${info.id}</p>
          <p class="search-name">${info.name}</p>
        </div>
      `;
      li.innerHTML = liContents;
      searchList.appendChild(li);
    }
  });
}
```

joonsikyang commented on students/14th/hocheolJang/style/main.css:
```diff
@@ -0,0 +1,502 @@
+  transform: rotate(-45deg);
+  position: relative;
+  top: 5px;
+}
```
- css 전체적으로 깔끔하게 잘 해주셨네요-! 
- float 속성도 보이는데 나중에 레이아웃 작업시 조정하기 어려운 부분들이 발생할 수 있어서 요즘엔 flex를 대부분 사용하는 추세입니다. 참고해주세요!
- css 요소 사이 사이는 한 줄 띄워주세요!

HocheolJang replied:
네 알겠습니다~
float보다는 flex를 사용할 수 있도록 하겠습니다~
감사합니다~

joonsikyang reviewed (commented):
호철님! 레이아웃 구현 깔끔하네요 :) 기능 구현 추가해서 push 다시 해주시고 리뷰 반영까지 해보세요👐

HocheolJang commented on students/14th/hocheolJang/login.html:
```diff
@@ -0,0 +1,33 @@
+    <link href="style/login.css" rel="stylesheet" type="text/css" />
+  </head>
+  <body>
+    <div class="box">
```
네 알겠습니다 ~ 수정하겠습니다!

HocheolJang commented on students/14th/hocheolJang/login.html:
```diff
@@ -0,0 +1,33 @@
+  <body>
+    <div class="box">
+      <div class="logo">
+        <img src="./img/logo_text.png" id="mainLogo" />
```
그 것은 바로 ~~~~~ A. L. T. 
alt값 추가하겠습니다!

HocheolJang commented on students/14th/hocheolJang/login.html:
```diff
@@ -0,0 +1,33 @@
+      <!-- <p>또는</p>
+      <p>Facebook으로 로그인</p> -->
+      <footer>
+        <a href="#" id="forgetPasswordbtn"><p id="forgetPassword">비밀번호를 잊으셨나요?</p></a>
```
음....
<p></p> 태그는 paragraph, 즉 문단의 약자로, 하나의 문단을 만들 때 쓰입니다.

아...문단을 만들 때 쓰는게 좋군요!
그럼 여기에는 span태그로 바꿔서 다시 해보면 더 좋겠죠? ㅎㅎㅎ

HocheolJang commented on students/14th/hocheolJang/main.html:
```diff
@@ -0,0 +1,243 @@
+<!DOCTYPE html>
+<html>
+
+<head>
+  <meta charset="utf-8" />
+  <meta name="viewport" content="width=device-width" />
+  <title>Main | Instagram</title>
+
+  <!-- CSS & Icon  -->
+  <link href="style/common.css" rel="stylesheet" type="text/css" />
+  <link href="style/main.css" rel="stylesheet" type="text/css" />
+  <script src="https://kit.fontawesome.com/8f5037fcbd.js" crossorigin="anonymous"></script>
+</head>
```
네 알겠습니다~ 이것도 붙여서 작성하겠습니다 👍

HocheolJang commented on students/14th/hocheolJang/main.html:
```diff
@@ -0,0 +1,243 @@
+          <div class="storyItem">
+            <img src="./img/icon/rainbow.png" class="storyProfileBackground" />
+            <img src="./img/profile/profile4.jpg" class="storyProfilePicture" />
+            <p class="storyAccount">kim_chong_in</p>
+          </div>
+          <div class="storyItem">
+            <img src="./img/icon/rainbow.png" class="storyProfileBackground" />
+            <img src="./img/profile/profile5.jpg" class="storyProfilePicture" />
+            <p class="storyAccount">500daysinsummer</p>
+          </div>
+          <div class="storyItem">
+            <img src="./img/icon/rainbow.png" class="storyProfileBackground" />
+            <img src="./img/profile/profile6.jpg" class="storyProfilePicture" />
+            <p class="storyAccount">polly_essay</p>
+          </div>
+          <div class="storyItem">
+            <img src="./img/icon/rainbow.png" class="storyProfileBackground" />
+            <img src="./img/profile/profile7.jpg" class="storyProfilePicture" />
+            <p class="storyAccount">joonhyukjang</p>
+          </div>
+          <div class="storyItem">
+            <img src="./img/icon/rainbow.png" class="storyProfileBackground" />
+            <img src="./img/profile/profile8.jpg" class="storyProfilePicture" />
+            <p class="storyAccount">junie.journey</p>
+          </div>
+          <div class="storyItem">
+            <img src="./img/icon/rainbow.png" class="storyProfileBackground" />
+            <img src="./img/profile/profile9.jpg" class="storyProfilePicture" />
+            <p class="storyAccount">kimjee__</p>
+          </div>
+          <div class="storyItem">
+            <img src="./img/icon/rainbow.png" class="storyProfileBackground" />
+            <img src="./img/profile/profile10.jpg" class="storyProfilePicture" />
+            <p class="storyAccount">heokyu0420</p>
+          </div>
+          <div class="storyItem">
+            <img src="./img/icon/rainbow.png" class="storyProfileBackground" />
+            <img src="./img/profile/profile11.jpg" class="storyProfilePicture" />
+            <p class="storyAccount">0_h__h_0</p>
+          </div>
```
넵 알겠습니다~ 리팩토링 할 때 다시 도전해보겠습니다 !!!!!

HocheolJang commented on students/14th/hocheolJang/main.html:
```diff
@@ -0,0 +1,243 @@
+              </div>
+            </div>
+          </div>
+          <div class="recommendationForMe">
```
네...이기적인 이름인 것같습니다 ㅋㅋㅋ
class명 더 고민해보겠습니다~~~

HocheolJang commented on students/14th/hocheolJang/main.html:
```diff
@@ -0,0 +1,243 @@
+              <span class="recommendationTitle">회원님을 위한 추천</span>
+              <a href="#" class="ShowAll"><span>모두 보기</span></a>
+              <div class="recommendationItemContainer">
+                <div class="recommendationItem">
+                  <img src="./img/profile/profile10.jpg" class="recommendationItemProfileImage" />
+                  <span class="recommendationAccount">heogue1210</span></br>
+                  <span class="recommendationdescription">yun_drum님 외 2명이 팔로우합니다</span>
+                  <a href="#" class="followBtn"><span>팔로우</span></a>
+                </div>
+                <div class="recommendationItem">
+                  <img src="./img/profile/profile10.jpg" class="recommendationItemProfileImage" />
+                  <span class="recommendationAccount">heogue1210</span></br>
+                  <span class="recommendationdescription">yun_drum님 외 2명이 팔로우합니다</span>
+                  <a href="#" class="followBtn"><span>팔로우</span></a>
+                </div>
+                <div class="recommendationItem">
+                  <img src="./img/profile/profile10.jpg" class="recommendationItemProfileImage" />
+                  <span class="recommendationAccount">heogue1210</span></br>
+                  <span class="recommendationdescription">yun_drum님 외 2명이 팔로우합니다</span>
+                  <a href="#" class="followBtn"><span>팔로우</span></a>
+                </div>
+                <div class="recommendationItem">
+                  <img src="./img/profile/profile10.jpg" class="recommendationItemProfileImage" />
+                  <span class="recommendationAccount">heogue1210</span></br>
+                  <span class="recommendationdescription">yun_drum님 외 2명이 팔로우합니다</span>
+                  <a href="#" class="followBtn"><span>팔로우</span></a>
+                </div>
+                <div class="recommendationItem">
+                  <img src="./img/profile/profile10.jpg" class="recommendationItemProfileImage" />
+                  <span class="recommendationAccount">heogue1210</span></br>
+                  <span class="recommendationdescription">yun_drum님 외 2명이 팔로우합니다</span>
+                  <a href="#" class="followBtn"><span>팔로우</span></a>
+                </div>
+                <div class="recommendationItem">
+                  <img src="./img/profile/profile10.jpg" class="recommendationItemProfileImage" />
+                  <span class="recommendationAccount">heogue1210</span></br>
+                  <span class="recommendationdescription">yun_drum님 외 2명이 팔로우합니다</span>
+                  <a href="#" class="followBtn"><span>팔로우</span></a>
+                </div>
```
넵 알겠습니다~ 공부해서 적용해볼 수 있도록 하겠습니다~

HocheolJang commented:
전체적으로 1차 리뷰해주셔서 감사합니다!
말씀해주신 부분 최대한 고쳐서 다시 리뷰요청 드리겠습니다!

wjdgotn77 committed:
[Add: first-commit westagram]

- html: login, main 작성
- style: login.css 작성 완료, common.css 작성중
- src: image files 추가

wjdgotn77 committed:
[Add: 메인 페이지 HTML,CSS 구현]

- src: 이미지 파일 추가
- main.html: 메인 페이지 구조 완성
- common.css: 메인 페이지 css 완성

wjdgotn77 committed:
[Add: 로그인 유효성검사 구현]

- login.js:
  1. 아이디, 비밀번호 입력시 로그인버튼 활성화
  2. 로그인버튼 클릭시 main.html로 이동

wjdgotn77 committed:
[Add: 엔터키 로그인 추가 구현]

wjdgotn77 committed:
[Add: 메인페이지 댓글기능 구현]

wjdgotn77 committed:
[Modify: 로그인 로직 수정]

wjdgotn77 committed:
[Refactor: makeComment 함수 수정 및 이벤트핸들링 함수 조건 추가]

makeComment 함수 수정
- createElement와 appendChild 사용 대신 innerHTML을 통해 더 간결하게 코드를 작성.

handleInput, handleClick 함수 조건 추가
- 댓글에 value가 입력되어야만 이벤트 실행되도록 조건 추가.

wjdgotn77 committed:
[Modify: 이미지태그 alt속성 위치 변경]

wjdgotn77 committed:
[Refactor: common.css파일 height 속성 리팩토링]

wjdgotn77 committed:
[Add: 로그인 페이지 아이디 및 비밀번호 유효성 검사]

- id : @가 포함되어야하는 아이디 조건 추가
- password : 12자리 이상 입력해야하는 패스워드 조건 추가

구현 사항
- id와 password 인풋에 각각 유효성 검사하는 함수추가
- 위의 조건이 충족 되었을 때 form의 submit가능한 함수 추가.

wjdgotn77 committed:
[Modify: 주석삭제 및 뒤로가기 시 input 비워주기]

wjdgotn77 committed:
[Add: common.css 댓글 삭제 버튼 추가]

wjdgotn77 committed:
[Add: 메인페이지 댓글 삭제 기능 구현]

wjdgotn77 committed:
[Add: 로그인 페이지 유효성검사 html 섹션 추가]

wjdgotn77 committed:
[Add: 로그인 페이지 유효성검사 css 추가]

wjdgotn77 committed:
[Add: 로그인 페이지 이벤트 종류 input으로 변경 및 className 수정]

wjdgotn77 committed:
[Delete: README.md 삭제]

wjdgotn77 committed:
[Add: 아이디 검색 기능 구현]

- 아이디를 담고 있는 배열 선언.
- 검색창에 텍스트 입력시 해당 텍스트와 일치하는 아이디가 리스트로 보임.
